sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -0,0 +1,358 @@
+        baseUri: uriOrConnectionString,
+        deserializationContentTypes
+      });
+    } else {
+      throw new Error(
+        "You must provide either a connection string or a URL and a ManagedIdentityCredential."
+      );
+    }
+  }
+
+  /**
+   * Add a setting into the Azure App Configuration service, failing if it
+   * already exists.
+   *
+   * Example usage:
+   * ```ts
+   * const result = await client.addConfigurationSetting({ key: "MyKey", label: "MyLabel", value: "MyValue" });
+   * ```
+   * @param configurationSetting A configuration setting.
+   * @param options Optional parameters for the request.
+   */
+  async addConfigurationSetting(
+    configurationSetting: ConfigurationSettingParam,
+    options: ConfigurationSettingOptions = {}
+  ): Promise<AddConfigurationSettingResponse> {
+    const result = await this.client.putKeyValue(configurationSetting.key, {
+      ifNoneMatch: "*",
+      label: configurationSetting.label,
+      entity: configurationSetting,
+      ...options
+    });
+
+    return result;
+  }
+
+  /**
+   * Delete a setting from the Azure App Configuration service
+   *
+   * Example usage:
+   * ```ts
+   * const deletedSetting = await client.deleteConfigurationSetting("MyKey", { label: "MyLabel" });
+   * ```
+   * @param key The name of the key.
+   * @param options Optional parameters for the request (ex: etag, label)
+   */
+  deleteConfigurationSetting(
+    key: string,
+    options: DeleteConfigurationSettingOptions & ETagOption
+  ): Promise<DeleteConfigurationSettingResponse> {
+    if (options.etag) {
+      options = { ...options };
+      options.ifMatch = formatETagForMatchHeaders(options);
+    }
+
+    return this.client.deleteKeyValue(key, options);
+  }
+
+  /**
+   * Gets a setting from the Azure App Configuration service.
+   *
+   * Example code:
+   * ```ts
+   * const setting = await client.getConfigurationSetting("MyKey", { label: "MyLabel" });
+   * ```
+   * @param key The name of the key.
+   * @param options Optional parameters for the request.
+   */
+  async getConfigurationSetting(
+    key: string,
+    options: GetConfigurationSettingOptions = {}
+  ): Promise<GetConfigurationSettingResponse> {
+    const result = await this.client.getKeyValue(key, {
+      label: options.label,
+      ...options
+    });
+
+    return result;
+  }
+
+  /**
+   * Lists settings from the Azure App Configuration service, optionally
+   * filtered by key names, labels and accept datetime.
+   *
+   * Example code:
+   * ```ts
+   * const allSettingsWithLabel = await client.listConfigurationSettings({ labels: [ "MyLabel" ] });
+   * ```
+   * @param options Optional parameters for the request.
+   */
+  listConfigurationSettings(
+    options: ListConfigurationSettingsOptions = {}
+  ): PagedAsyncIterableIterator<ConfigurationSetting, ListConfigurationSettingPage> {
+    const iter = this.getListConfigurationSettingsIterator(options);
+
+    return {
```

Comment:
are we just wrapping this other iterator because of the page size thing? or are they different types?

Reply:
From what I can tell they're just for different use cases.

The "iterate over individual items" one is probably the one most people would like to use since it allows you to just iterate over a magical collection of ConfigurationSettings without having to manage or see HTTP related state.

The "iterate by page" iterator is convenient since it does the 'nextLink' logic for you _but_ you can also now see the returned HTTP responses.

Otherwise they're the same underlying method, just exposed two different ways.